File: Pony_Racing/core/src/com/nopalsoft/ponyrace/game_objects/BloodStone.java
package com.nopalsoft.ponyrace.game_objects;

import java.util.Random;

public class BloodStone extends GameObject {
    public float lastStateTime;
    public float stateTime;
    public Type type;

    public BloodStone(float x, float y, Type type, Random random) {
        super(x, y);
        stateTime = random.nextFloat() * 5f;
        lastStateTime = stateTime;
        this.type = type;
    }

    public void update(float delta) {
        lastStateTime = stateTime;
        stateTime += delta;
    }

    public enum Type {
        SMALL, MEDIUM, LARGE
    }
}

File: Pony_Racing/core/src/com/nopalsoft/ponyrace/game_objects/BloodStone.kt
package com.nopalsoft.ponyrace.game_objects

import java.util.Random

class BloodStone(x: Float, y: Float, val type: Type, random: Random) : BaseGameObject(x, y) {

    var stateTime = random.nextFloat() * 5F
    var lastStateTime = stateTime

    fun update(delta: Float) {
        lastStateTime = stateTime
        stateTime += delta
    }

    enum class Type {
        SMALL, MEDIUM, LARGE
    }
}